test/unittest/StatisticsBackend/IsActiveTests.cpp

```diff
@@ -0,0 +1,474 @@
+        eprosima::fastrtps::rtps::WriterProxyData data(1, 1);
+
+        // The discovered writer is in the participant
+        eprosima::fastrtps::rtps::GUID_t writer_guid_;
+        std::stringstream(datawriter->guid) >> writer_guid_;
+        data.guid(writer_guid_);
+
+        // The discovered writer is in the topic
+        data.topicName(topic->name);
+        data.typeName(topic->data_type);
+
+        // The discovered writer contains the locator
+        eprosima::fastrtps::rtps::Locator_t dds_existing_unicast_locator(LOCATOR_KIND_UDPv4, 1024);
+        dds_existing_unicast_locator.address[12] = 127;
+        dds_existing_unicast_locator.address[15] = 1;
+        data.add_unicast_locator(dds_existing_unicast_locator);
+
+        // Finish building the discovered writer info
+        eprosima::fastrtps::rtps::WriterDiscoveryInfo info(data);
+        info.status = eprosima::fastrtps::rtps::WriterDiscoveryInfo::REMOVED_WRITER;
+
+        // Execution: Call the listener
+        participant_listener.on_publisher_discovery(&statistics_participant, std::move(info));
+    }
+
+    // Reader
+    {
+        // Start building the discovered reader info
+        eprosima::fastrtps::rtps::ReaderProxyData data(1, 1);
+
+        // The discovered reader is in the participant
+        eprosima::fastrtps::rtps::GUID_t reader_guid_;
+        std::stringstream(datareader->guid) >> reader_guid_;
+        data.guid(reader_guid_);
+
+        // The discovered reader is in the topic
+        data.topicName(topic->name);
+        data.typeName(topic->data_type);
+
+        // The discovered reader contains the locator
+        eprosima::fastrtps::rtps::Locator_t dds_existing_unicast_locator(LOCATOR_KIND_UDPv4, 1024);
+        dds_existing_unicast_locator.address[12] = 127;
+        dds_existing_unicast_locator.address[15] = 1;
+        data.add_unicast_locator(dds_existing_unicast_locator);
+
+        // Finish building the discovered reader info
+        eprosima::fastrtps::rtps::ReaderDiscoveryInfo info(data);
+        info.status = eprosima::fastrtps::rtps::ReaderDiscoveryInfo::REMOVED_READER;
+
+        // Execution: Call the listener
+        participant_listener.on_subscriber_discovery(&statistics_participant, std::move(info));
+    }
+
+    ASSERT_TRUE(StatisticsBackendTest::is_active(host->id));
+    ASSERT_TRUE(StatisticsBackendTest::is_active(user->id));
+    ASSERT_TRUE(StatisticsBackendTest::is_active(process->id));
+    ASSERT_TRUE(StatisticsBackendTest::is_active(domain->id));
+    ASSERT_FALSE(StatisticsBackendTest::is_active(topic->id));
+    ASSERT_TRUE(StatisticsBackendTest::is_active(participant->id));
+    ASSERT_FALSE(StatisticsBackendTest::is_active(datawriter->id));
+    ASSERT_FALSE(StatisticsBackendTest::is_active(datareader->id));
+    ASSERT_TRUE(StatisticsBackendTest::is_active(locator->id));
+}
+
+// Check the is_active StatisticsBackend method when the dds_entities are undiscovered
+TEST_F(is_active_tests, dds_entities)
```

Comment:
I believe this test can be removed, the cases are already covered in the tests for participant and endpoints in isolation

Reply:
Like I say before about `DataseStatusTests`, confirm me that is not necessary please.

Reply:
Same response as before

Reply:
Same response as before